### PR TITLE
Make generated CoreData model class final when model isn't abstract

### PR DESCRIPTION
**Pull request by grdsdev**
Change Core Data Swift 5 template for adding support to set a class as `final` when the model isn't abstract.

### CODE REVIEW

**Comment by grdsdev** [comment]
What you guys think about this? @djbe and @AliSoftware

File: Sources/SwiftGenCLI/templates/coredata/swift5.stencil [hunk 1]
```diff
@@ -20,6 +20,7 @@ import {{ import }}
 {% for name, entity in model.entities %}
 {% set superclass %}{{ model.entities[entity.superEntity].className|default:"NSManagedObject" }}{% endset %}
 {% set entityClassName %}{{ entity.className|default:"NSManagedObject" }}{% endset %}
+{% set final %}{% if entity.isAbstract %}{% else %}final {% endif %}{% endset %}
```
Comment (djbe) [inline Sources/SwiftGenCLI/templates/coredata/swift5.stencil]: Avoid setting a property if you're only going to use it once, especially for for simple checks.

File: Sources/SwiftGenCLI/templates/coredata/swift5.stencil [hunk 2]
```diff
@@ -32,7 +33,7 @@ import {{ import }}
 {% if param.generateObjcName %}
 @objc({{ entityClassName }})
 {% endif %}
-{{ accessModifier }} class {{ entityClassName }}: {{ superclass }} {
+{{ accessModifier }} {{ final }}class {{ entityClassName }}: {{ superclass }} {
```
Comment (djbe) [inline Sources/SwiftGenCLI/templates/coredata/swift5.stencil]:
```suggestion
{{ accessModifier }} {% if not entity.isAbstract %}final {% endif %}class {{ entityClassName }}: {{ superclass }} {
```

**Review by djbe — CHANGES_REQUESTED** [review]
Good idea. I've added some comments on your changes.

Also note that this change needs to also be made for the other templates (`swift4` in this case), and generate the output as well for those.

**Comment by grdsdev** [comment]
Hey, @djbe I noticed that you already made the changes. Is there anything that still needs to be handled?

**Comment by djbe** [comment]
I was waiting for the tests to finish, and they curiously enough revealed an error. In our tests we have:
- A `MainEntity`, which is *not* abstract
- A `ChildEntity`, which has `MainEntity` as a parent

You can create this situation in the model editor. I have no idea how useful it is, or valid, but it's possible. Hence, we have to handle it gracefully.

**Comment by djbe** [comment]
There, that should solve it. Had to resolve to a trick in Stencil to check if something exists in an array 😅

**Review by djbe — APPROVED** [review]

**Pull request merged**